packages/nextjs-mf/src/plugins/container/AsyncBoundaryPlugin.ts

```diff
@@ -0,0 +1,163 @@
+
+/**
+ * AsyncBoundaryPlugin is a Webpack plugin that handles asynchronous boundaries in a federated module.
+ * @class
+ */
+class AsyncBoundaryPlugin {
+  /**
+   * Apply the plugin to the Webpack compiler instance.
+   * @param {Compiler} compiler - Webpack compiler instance.
+   */
+  public apply(compiler: Compiler): void {
+    const { javascript } = compiler.webpack;
+    compiler.hooks.thisCompilation.tap(
+      'AsyncBoundaryPlugin',
+      (compilation: Compilation) => {
+        const hooks =
+          javascript.JavascriptModulesPlugin.getCompilationHooks(compilation);
+        //@ts-ignore
+        hooks.renderStartup.tap(
+          'AsyncBoundaryPlugin',
+          (
+            source: Source,
+            renderContext: Module,
+            startupRenderContext: StartupRenderContext,
+          ) => {
+            return this.renderStartupLogic(
+              source,
+              renderContext,
+              startupRenderContext,
+              compilation,
+            );
+          },
+        );
+      },
+    );
+  }
+
+  /**
+   * Render the startup logic for the plugin.
+   * @param {Source} source - The source code.
+   * @param {RenderContext} renderContext - The render context.
+   * @param {any} startupRenderContext - The startup render context.
+   * @param {Compilation} compilation - The Webpack compilation instance.
+   * @returns {string} - The modified source code.
+   */
+  private renderStartupLogic(
+    source: Source,
+    renderContext: Module,
+    startupRenderContext: StartupRenderContext,
+    compilation: Compilation,
+  ): string {
+    const isInvalidContext = this.checkInvalidContext(
+      renderContext,
+      compilation,
+    );
+    if (isInvalidContext) return source.source().toString();
+
+    const { chunkGraph } = compilation;
+    const replaceSource = source.source().toString();
+    const [webpack_exec, ...webpack_exports] = replaceSource.split('\n');
+    const dependentChunkIds = this.getDependentChunkIds(
+      startupRenderContext,
+      chunkGraph,
+    );
+
+    return Template.asString([
+      this.replaceWebpackExec(webpack_exec),
+      'globalThis.ongoingRemotes = globalThis.ongoingRemotes || [];',
+      'var __webpack_exec__ = async function() {',
+      Template.indent([
+        `var chunkIds = ${JSON.stringify(Array.from(dependentChunkIds))};`,
+        `chunkIds.forEach((id) => ${RuntimeGlobals.ensureChunkHandlers}.consumes(id, globalThis.ongoingRemotes));`,
+        `await Promise.all(globalThis.ongoingRemotes);`,
+        'console.log("waiting for consume", globalThis.ongoingRemotes.length);',
+        `chunkIds.forEach((id) => ${RuntimeGlobals.ensureChunkHandlers}.remotes(id, globalThis.ongoingRemotes));`,
+        `await Promise.all(globalThis.ongoingRemotes);`,
+        'console.log("waiting for remote", globalThis.ongoingRemotes.length);',
+        `return  __original_webpack_exec__.apply(this, arguments);`,
```

Comment:
This is a little dirty, string replacing some of the generated startup of the entry runtime module. The part that does this natively is a few plugins deep in webpack, don’t want to copy over entry runtime module right now and there’s likely a slightly better way to do this. But the primitive rename of webpack exec and manual chunk handler is simple but  effective, can be improved later